### PR TITLE
Couple more reaction changes for Hosting

### PR DESCRIPTION
**Pull request by BrennanConroy**

### CODE REVIEW

File: test/Microsoft.AspNetCore.Sockets.Tests/MapEndPointTests.cs [hunk 1]
```diff
@@ -62,7 +65,7 @@ public void MapEndPointFindsAuthAttributeOnEndPoint()
         public void MapEndPointFindsAuthAttributeOnInheritedEndPoint()
         {
             var authCount = 0;
-            var builder = new WebHostBuilder()
+            using (var builder = new WebHostBuilder()
```
Comment (moozzyk) [inline test/Microsoft.AspNetCore.Sockets.Tests/MapEndPointTests.cs]: does it make sense to move building the host to a separate function?

Comment (BrennanConroy) [reply test/Microsoft.AspNetCore.Sockets.Tests/MapEndPointTests.cs]: They all have slightly different services and other settings

Comment (muratg) [reply test/Microsoft.AspNetCore.Sockets.Tests/MapEndPointTests.cs]: Sounds reasonable.

**Comment by BrennanConroy** [comment]
🆙 📅

**Comment by davidfowl** [comment]
Whats the point of this change?

**Comment by BrennanConroy** [comment]
To make the tests pass. Hosting now delays config methods until start has been called

File: test/Microsoft.AspNetCore.SignalR.Tests/MapSignalRTests.cs [hunk 2]
```diff
@@ -14,17 +14,7 @@ public void MapSignalRFailsForInvalidHub()
...
-                        .ConfigureServices(services =>
-                        {
-                            services.AddSignalR();
-                        })
-                        .Configure(app =>
-                        {
-                            app.UseSignalR(options => options.MapHub<InvalidHub>("overloads"));
-                        })
-                        .Build())
+                using (var builder = BuildWebHost(options => options.MapHub<InvalidHub>("overloads")))
```
Comment (JanEggers) [inline test/Microsoft.AspNetCore.SignalR.Tests/MapSignalRTests.cs]: shouldt this var be named host

**Pull request merged**